src/scheduler/annualReviewStartEmailScheduler.ts

```diff
@@ -0,0 +1,211 @@
+import { findListByAnnualReviewDate } from "server/models/list";
+import {
+  List, ListItemGetObject,
+} from "server/models/types";
+import { logger } from "server/services/logger";
+import { findListItemsForLists, updateAnnualReview } from "server/models/listItem";
+import {
+  sendAnnualReviewPostEmail, sendAnnualReviewProviderEmail,
+} from "server/services/govuk-notify";
+import { BaseDeserialisedWebhookData } from "server/models/listItem/providers/deserialisers/types";
+import { initialiseFormRunnerSession } from "server/components/formRunner/helpers";
+import { lowerCase, startCase } from "lodash";
+import { AuditEvent, ListItem, ListItemEvent, Status } from "@prisma/client";
+import { addDays } from "date-fns";
+import { recordListItemEvent } from "server/models/audit";
+import { SCHEDULED_PROCESS_TODAY_DATE } from "server/config";
+import { getTodayDate } from "./helpers";
+import { EventMetaData } from "server/utils/validation";
+
+const ONE_MONTH_AWAY = 28;
+const ONE_WEEK_AWAY = 7;
+const ONE_DAY_AWAY = 1;
+
+async function addAuditEventForEmail(context: AnnualReviewStartDateContext, list: List) {
+  const postEvents: Record<number, EventMetaData> = {
+    0: {
+      auditEvent: AuditEvent.ANNUAL_REVIEW_STARTED_POST_EMAIL_SENT,
+      auditListItemEventName: "sendAnnualReviewStartedPostEmail",
+    },
+    1: {
+      auditEvent: AuditEvent.ANNUAL_REVIEW_START_ONE_DAY_REMINDER_POST_EMAIL_SENT,
+      auditListItemEventName: "sendAnnualReviewOneDayReminderPostEmail",
```

Comment:
if they were just variables I'd agree but these are essentially values that are used in the Audit table so the full description would make more sense here unless you default them all to EventType value "ReminderSent" and introduce another column "Reminder" to the Audit table which seems a bit heavy.  

Perhaps a better way is to create a new table that will act as a scheduled process email audit and have the scheduled process persist a record there with isSent = false and a separate worker to read the table. and set.isSent = true which will facilitate a more robust retry capability. This will then remove the need for these AuditEvent entries. 
 Similar to what we discussed previously.

Reply:
You can use the json field to indicate failure. 

Would it be better to have a generic reminder enum type, and have a more descriptive json?

Reply:
FWIW, enums are quite inflexible. Removing a value will break the DB unless you migrate all of the existing values.

Reply:
Although yes.. I think batch (into table), then worker to just exec those lines is nicer in terms of support.